eskipfile/remote.go

```diff
@@ -0,0 +1,146 @@
+package eskipfile
+
+import (
+	"errors"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/routing"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type remoteEskipFile struct {
+	preloaded       bool
+	remotePath      string
+	localPath       string
+	eskipFileClient *WatchClient
+	threshold       int
+	verbose         bool
+}
+
+// RemoteWatch creates a route configuration client with (remote) file watching. Watch doesn't follow file system nodes,
+// it always reads (or re-downloads) from the file identified by the initially provided file name.
+func RemoteWatch(rf string, th int, v bool) (routing.DataClient, error) {
+	if !isFileRemote(rf) {
+		return Watch(rf), nil
```

Comment:
instead of falling back to a local file in each case when it's not remote, how about doing the following:

- http:/https: -> use http(s)
- file: -> use local file
- no schema prefix -> use local file
- unknown schema prefix -> error

Reply:
also, I would move this decision logic outside of the `RemoteWatch` initializer, because this way it is a bit misleading

Reply:
I actually was initially considering doing this logic in the `Watch` dataClient, and have a private `localWatch` and a private `remoteWatch` data clients, so: 
1. it's could be done so it's backwards compatible
2. a consolidated API for Watched files, regardless where they are.

But I didn't do it in case it felt too "disrupting". But if you like the approach, I'm happy to give another thought.